server/lib/nl/detection/llm_fallback.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Decide whether to fallback to using LLM."""
+
+from typing import List
+
+from server.lib.nl.common import counters
+from server.lib.nl.detection import utils
+from server.lib.nl.detection.types import ClassificationType
+from server.lib.nl.detection.types import Detection
+from shared.lib import detected_variables as vars
+
+_COMPLEX_QUERY_TOKEN_THRESHOLD = 8
+
+
+#
```

Comment:
For ease of future updates, good to list all the "llm" query examples in the function comment.

Reply:
Done.